Load user mesh config if set and display on the mesh page.

nrfox opened the pull request:
### Describe the change

Respects the `SHARED_MESH_CONFIG` setting when autodetecting istiod configuration. This configuration will be properly merged with the existing standard configmap. The mesh page will show a tab per configuration so you can compare the differences between each along with the effective configuration after each has been merged.

### Steps to test the PR

1. Deploy istio and Kiali
    `hack/run-integration-tests.sh --test-suite backend --setup-only true`
1. Create a user mesh config with some mesh config setting:
    ```
    apiVersion: v1
    data:
      mesh: |-
        disableEnvoyListenerLog: true
    kind: ConfigMap
    metadata:
      name: istio-user
      namespace: istio-system
    
    ```
1. Set the `SHARED_MESH_CONFIG` istiod env var
    ```
    kubectl edit istios default
    ```
    ```
    spec:
      values:
        pilot:
          env:
            SHARED_MESH_CONFIG: istio-user
    ```

1. Navigate to the mesh page and click on istiod node.
    You should see 3 tabs for each configuration along with a help icon to get more info about the config.
    ![Screenshot from 2025-04-11 13-18-23](https://github.com/user-attachments/assets/fc25f2a2-890f-4a4c-9ad9-8ee9ed034b0e)

    
### Automation testing

Unit tests and e2e tests included.

### Issue reference

https://github.com/kiali/kiali/issues/8248

nrfox commented:
@jshaughn @ferhoyos I'm unsure of how I feel about the help icon in the tabs. I like that it shows more info about the configmap but maybe the config map name/namespace/cluster should go in the card itself around where "configuration" is today. Can maybe keep the help icon to explain what each tab means. WDYT?

ferhoyos commented:
I think there is enough space to put the config map name/namespace/cluster information in the card and let the help icon to explain what each tab means. @fkargbo can provide some input here.

fkargbo commented:
Let's review this in our next meeting for context. Thanks! @nrfox @ferhoyos

nrfox commented:
![shared_mesh_config](https://github.com/user-attachments/assets/dfb41d69-068d-4653-be69-59504e8ee585)

@ferhoyos In addition to the shared mesh config changes, I changed some of the formatting on the summary panel to be more consistent:
1. Added titles to certs/metrics page
2. Removed info icons in favor of just hovering over the text to get the tooltip to appear
3. Min TLS version now shows `TLSV1_2` when `TLS_AUTO` is specified instead of `N/A`. `TLSV1_2` is more accurate since this is the default.

ferhoyos commented:
Thanks @nrfox, I like your changes, it looks more consistent now. My only concern is removing the info icons in favor of just hover text. Without any visual cues, users won't know which texts have tooltips and will probably find them by mistake. Hovering over text isn't a natural action if there's no indication of a tooltip. @fkargbo what do you think?

nrfox commented:
> My only concern is removing the info icons in favor of just hover text. Without any visual cues, users won't know which texts have tooltips and will probably find them by mistake.

@ferhoyos yeah I wasn't totally sure about this but thought that tooltip was for giving more context to an icon and the "blue capsules" that previously had info icons on them we're more like icons themselves rather than just text so they don't need an additional info icon on them. I also removed the text from the `Memory` and `CPU` chart legend and those are just text so probably I shouldn't have removed the info icon there and I'll add it back.

nrfox commented:
@ferhoyos I added back the tooltip for the graph but left the info icon off of the capsules since those functionally look like icons to me.

fkargbo commented:
/label ux-approved

jshaughn reviewed (approved):
Code looks fine and includes tests, some better typing, and a fix for TLS, all good.  working great:
[Screencast From 2025-05-16 12-26-56.webm](https://github.com/user-attachments/assets/70da3fe0-e209-4ec7-ba39-b46d20e4a0d3)

This did not get demoed in the sprint demo, so maybe a demo next time, or a short blog, it's a good feature and we need to make sure it's marketed!

nrfox commented:
> This did not get demoed in the sprint demo, so maybe a demo next time, or a short blog, it's a good feature and we need to make sure it's marketed!

Yeah I wasn't sure if it would make it into this sprint or not. I'll do a demo next time.

ferhoyos commented on frontend/src/components/MTls/MeshMTLSStatus.tsx:
```diff
@@ -94,6 +98,7 @@ class MeshMTLSStatusComponent extends React.Component<Props> {
     // leaving empty cluster param here, home cluster will be used by default
     API.getMeshTls(this.props.cluster, this.props.revision)
       .then(response => {
+        console.log(response);
```
We should not have console.log in the code, use debug logger if we have to print this value for debugging.

ferhoyos commented on frontend/cypress/integration/common/hooks.ts:
```diff
@@ -157,3 +157,39 @@ After({ tags: '@clean-istio-namespace-resources-after' }, () => {
   cy.exec('kubectl -n istio-system delete Sidecar default', { failOnNonZeroExit: false });
 });
+
+Before({ tags: '@shared-mesh-config' }, () => {
```
@nrfox can you please add this piece of code to OSSMC? The plugin has its own cypress hooks file.

ferhoyos reviewed (commented):
From my side, just a pair of comments in the code. In my opinion, it's fine to use 'em' but I prefer to use 'rem' because it is based on the root font and it is usually more consistent.

jshaughn commented:
@nrfox Please follow-up on the post-merge comments from @ferhoyos .... Thanks.
